test/lib/ansible_test/_internal/ci/shippable.py

```diff
@@ -184,6 +191,20 @@ def _get_merged_commit(self, args, commit):  # type: (CommonConfig, str) -> t.Op
         return last_commit
 
 
+class ShippableAuthHelper(OpenSSLAuthHelper):
+    """
+    Authentication helper for Shippable.
+    Based on OpenSSL since cryptography is not provided by the default Shippable environment.
+    """
+    def publish_public_key(self, public_key_pem_bytes):  # type: (bytes) -> None
+        """Publish the given public key."""
+        public_key_pem_string = public_key_pem_bytes.decode()
```

Comment:
Is `public_key_pem_bytes` just meant to be the base64 encoded value, if so then shouldn't this function take in a text string and not bytes?

Reply:
Yes, it is just the PEM encoded key, such as:

```
-----BEGIN PUBLIC KEY-----
MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAEcYXOnHJgcLaFzWs1j96C9lWGpyhkaVGK
XpvCmLc+IXFN7i66HQ+UGZNG9olB/nRDhg16HV6cgCBx+g1zhpvbSS+yl13cbuoP
wH/yU7v06UAO/FDH5cJ612onOiP8TpMt
-----END PUBLIC KEY-----
```

So it could be passed around as a string instead of bytes.

Reply:
I'll look at refactoring the code to use text internally.

Reply:
I've refactored the code to use text instead of bytes in most places. The `sign_bytes` function consumes and returns bytes, since that's what the signature generation uses.